src/common/mfu_flist.h

```diff
@@ -181,23 +181,26 @@ void mfu_flist_free(mfu_flist* flist);
 void mfu_flist_walk_path(
     const char* path,           /* IN  - path to be walked */
     mfu_walk_opts_t* walk_opts, /* IN  - functions to perform during the walk */
-    mfu_flist flist             /* OUT - flist to insert walked items into */
+    mfu_flist flist,            /* OUT - flist to insert walked items into */
+    mfu_file_t* mfu_file        /* IN  - I/O filesystem functions to use during the walk */
 );
 
 /* create file list by walking list of directories */
 void mfu_flist_walk_paths(
     uint64_t num_paths,         /* IN  - number of paths in array */
     const char** paths,         /* IN  - array of paths to be walkted */
     mfu_walk_opts_t* walk_opts, /* IN  - functions to perform during the walk */
-    mfu_flist flist             /* OUT - flist to insert walked items into */
+    mfu_flist flist,            /* OUT - flist to insert walked items into */
+    mfu_file_t* mfu_file        /* IN  - I/O filesystem functions to use during the walk */
```

Comment:
Some of these functions take a list of source paths.  Will we need a corresponding mfu_file for each?  Or do we want to restrict all source paths to be of the same type?  Perhaps to start, we could restrict to one type, but in that case, we should probably run a check and kick out with an error if not.  I haven't looked at the whole PR yet, just thinking out loud...